src/badguy/bouncing_snowball.cpp

```diff
@@ -69,25 +70,20 @@ BouncingSnowball::collision_squished(GameObject& object)
 void
 BouncingSnowball::collision_solid(const CollisionHit& hit)
 {
-  if (m_sprite->get_action() == "squished")
-  {
+  if (m_sprite->get_action() == "squished") {
     return;
   }
 
-  if (hit.bottom) {
-    if (get_state() == STATE_ACTIVE) {
-      float bounce_speed = -m_physic.get_velocity_y()*0.8f;
-      m_physic.set_velocity_y(std::min(JUMPSPEED, bounce_speed));
-	    m_sprite->set_action(m_dir == Direction::LEFT ? "left-up" : "right-up", /* loops = */ 1);
-    } else {
-      m_physic.set_velocity_y(0);
-    }
-  } else if (hit.top) {
+  if (hit.bottom && get_state() == STATE_ACTIVE) {
+    float bounce_speed = -m_physic.get_velocity_y()*0.8f;
+    m_physic.set_velocity_y(std::min(JUMPSPEED, bounce_speed));
+    m_sprite->set_action(m_dir == Direction::LEFT ? "left-up" : "right-up", /* loops = */ 1);
+  } else {
     m_physic.set_velocity_y(0);
   }
```

Comment:
Good refactor :+1: 

The :b:'s though :stuck_out_tongue: 

EDIT:
`hit.top`and `hit.bottom` are not opposite variables; they can be both false or both true at the same time. Make sure your refactor takes that into consideration!